### ISSUE TITLE
Trying to query without prior connect should throw an error

### ISSUE THREAD

**Issue by fcFn** [brianc/node-postgres#1861]
I forgot to connect with `client.connect()`, and spent more time than I should've when my queries silently failed.

Should an error be thrown when trying to use `query` without prior `connect`?

If so, I could probably whip up a PR.

Thanks!

**Comment by vitaly-t** [comment]
If you follow the recommended approach of using `query` on the connection object from the pool, then it becomes impossible to make a query before getting the connection first.

And what you're referring to is the obsolete approach to using the global `query` method, which you should no longer use anyhow.

**Comment by fcFn** [comment]
I know I should've read the whole documentation (mea culpa), but the [Welcome section](https://node-postgres.com/#getting-started) of the docs uses the global `query` method in the examples and it doesn't mention that it is deprecated. I understand that a Hello world example should be as simple as possible, yet I don't think it should use obsolete approaches.

**Comment by charmander** [comment]
`pg.connect` and `pg.query` are already gone. `Client#query` isn’t deprecated. The client has a query queue that lets queries be issued before connecting, but I think it’s a good idea to remove that.

**Comment by acco** [comment]
This burned me pretty bad today.

It appears if you call `client.query()` before you call `client.connect()`, this queues up a query to be executed whenever you may happen to call `client.connect()` in the future. If you forgot to call `client.connect()` (an honest mistake), this means you get back a promise that just never resolves.

This should really throw an error though. There is no connection to perform a query over. The intended feature of enqueuing queries to be performed after an upcoming connection event should reside under a different method, like `enqueueQuery()`. It's asking the client to perform a different sort of job. And I have to imagine this feature is only rarely used as intended.

**Comment by brianc** [comment]
I agree the query queue is weird....it's _very_ old.  I think I based it on the original design of the mysql lib which has also undergone extensive changes throughout the years.  I'm in favor of removing it as well...it seemed like a good idea at the time, but it wasn't.  It's just a weird artifact from long ago.  Obviously a breaking change to remove it, but probably wont bite that much since like you I doubt many people use it other than accidentally.
